include/verilated.h

```diff
@@ -252,6 +253,26 @@ class VerilatedAssertOneThread final {
 #endif
 };
 
+//=========================================================================
+/// Base class of a Verilated model.
+
+class VerilatedModel VL_NOT_FINAL {
+    VL_UNCOPYABLE(VerilatedModel);
+
+protected:
+    explicit VerilatedModel() = default;
+    virtual ~VerilatedModel() = default;
+
+public:
+    /// Returns the hierarchical name of this module instance.
+    virtual const char* hierName() = 0;
+    /// Returns the name of this model (the name of the generated model class).
+    virtual const char* modelName() = 0;
+    /// Returns the thread level parallelism (the value of --threads), this model was Verilated
+    /// with. Always 1 or larger.
+    virtual unsigned threads() = 0;
```

Comment:
I think overloading properties so out of range values take on magic meaning is pretty poor interface design. I could rename this to "parallelism" instead but that is its intended use, ie: tell me how many hardware threads you want. I don't want to have to max(1, _) this everywhere it's used. Also I think the model itself is thread safe with threads 0, but libverilated isn't. It would be unfortunate to blur the boundary between model and runtime library again.

Reply:
Ok, fine with this then.